pkg/apis/azuredisk/v1beta2/types.go

```diff
@@ -521,3 +521,88 @@ type ListSnapshotsResult struct {
 	// +optional
 	NextToken string `json:"next_token,omitempty"`
 }
+
+type AzDiskDriverConfiguration struct {
+	ControllerConfig ControllerConfiguration `json:"controller"`
```

Comment:
You can make all of the field `omitempty` to make them optional in the `ConfigMap`.